src/uu/rm/src/rm.rs

```diff
@@ -87,17 +85,34 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(|v| v.map(ToString::to_string).collect())
         .unwrap_or_default();
 
-    let force = matches.get_flag(OPT_FORCE);
+    let force_flag = matches.get_flag(OPT_FORCE);
+
+    // If -f(--force) is before any -i (or variants) we want prompts else no prompts
+    let force_prompt_never: bool = force_flag && {
+        if matches.value_source(OPT_FORCE) == Some(ValueSource::CommandLine) {
```

Comment:
This is already guaranteed by checking `force_flag`, but if you think that the explicitness is important we can leave it like this.

Reply:
You are right